nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/buffer/BaseCudaDataBuffer.java

```diff
@@ -949,6 +1123,91 @@ public void set(double[] data, long length, long srcOffset, long dstOffset) {
                 val pointer = new ShortPointer(ArrayUtil.toHalfs(data));
                 copyDataFromSrc(pointer,length,offset,dstOffset);
 
+            }
+            break;
+
+            case BFLOAT16: {
+                val pointer = new ShortPointer(ArrayUtil.toBfloats(data));
+                copyDataFromSrc(pointer,length,offset,dstOffset);
+
+            }
+            break;
+            case FLOAT: {
+                val pointer = new FloatPointer(ArrayUtil.toFloats(data));
+                copyDataFromSrc(pointer,length,offset,dstOffset);
+
+            }
+            break;
+            case DOUBLE: {
+                val pointer = new DoublePointer(ArrayUtil.toDouble(data));
+                copyDataFromSrc(pointer,length,offset,dstOffset);
+
+            }
+            break;
+            default:
+                throw new UnsupportedOperationException("Unsupported data type: " + dataType());
+        }
+
+
+    }
+
+    /**
+     *
+     * PLEASE NOTE: length, srcOffset, dstOffset are considered numbers of elements, not byte offsets
+     *
+     * @param data
+     * @param length
+     * @param srcOffset
+     * @param dstOffset
+     */
+    public void set(double[] data, long length, long srcOffset, long dstOffset) {
+        switch (dataType()) {
+            case BOOL:  {
+                val pointer = new BytePointer(ArrayUtil.toBytes(data));
+                copyDataFromSrc(pointer,length,offset,dstOffset);
+
+            }
+            break;
+
+            case UBYTE: {
+                data = ArrayUtil.cutBelowZero(data);
+            }
```

Comment:
Is this explicitly meant to fall through to the next case? If so, put a note here.